packages/flutter_tools/lib/src/dart/pub.dart

```diff
@@ -477,17 +529,51 @@ class _DefaultPub implements Pub {
     return values.join(':');
   }
 
-  String? _getRootPubCacheIfAvailable() {
+  /// There is 3 ways to get the cache path
+  /// 1) Provide the _kPubCacheEnvironmentKey.
+  /// 2) localPath (flutterRoot) but not globalPath ($HOME) .pubCache in which case we use the available (and vice versa).
+  /// 3) If both local and global are available joinCaches is called and local deleted
+  String? _getLocalPubCacheIfAvailable() {
     if (_platform.environment.containsKey(_kPubCacheEnvironmentKey)) {
       return _platform.environment[_kPubCacheEnvironmentKey];
     }
 
-    final String cachePath = _fileSystem.path.join(Cache.flutterRoot!, '.pub-cache');
-    if (_fileSystem.directory(cachePath).existsSync()) {
-      _logger.printTrace('Using $cachePath for the pub cache.');
-      return cachePath;
+    final String localCachePath = _fileSystem.path.join(Cache.flutterRoot!, '.pub-cache');
+    final Directory globalDirectory;
+    if (_platform.isWindows) {
+      // %LOCALAPPDATA% is preferred as the cache location over %APPDATA%, because the latter is synchronised between
```

Comment:
@jonasfj I copy paste this comment from the dart-lang repo. Also can you double check on this logic for windows? Thanks.